frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +260,62 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    const filePath = path.join('/', key);
+    try {
+      // get ml-pipeline-ui pod name
+      const { HOSTNAME: POD_NAME } = process.env;
+      // only check when POD_NAME and serverNamespace can be obtained
+      if (POD_NAME && serverNamespace) {
+        const [pod, err] = await k8sHelper.getPod(POD_NAME, serverNamespace);
+        if (err) {
+          const { message, additionalInfo } = err;
+          console.error(message, additionalInfo);
+          res.status(500).send(message);
+          return;
+        }
+
+        if (!pod) {
+          const message = `Could not get pod ${POD_NAME} in namespace ${serverNamespace}`;
+          res.status(500).send(message);
+          return;
+        }
+
+        // volumes not specified or volume named ${bucket} not specified
+        if (!pod.spec.volumes || !pod.spec.volumes.find(v => v.name === bucket)) {
```

Comment:
nit: `!Array.isArray(pod?.spec?.volumes) || pod.spec.volumes.find(v => v?.name === bucket)`
this will be 100% safe

Reply:
Also, can you look at the volumeMount object instead? It will have information about where it was mounted.